common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -631,4 +668,44 @@ private static void setCorrelationIdOnResult(@NonNull final CommandResult comman
         }
     }
 
+
+    private static void performRefresh(SilentTokenCommand command) {
+        SilentTokenCommandParameters params = (SilentTokenCommandParameters) command.getParameters();
```

Comment:
Would it be perhaps easier to keep the refresh_in logic within the acquireTokenSilent in LocalMSALController.java within a single command? If it's possible, this could simplify your changes.

Reply:
We discussed this too, but it was concluded that it'd be more minimal if we just reused the SilentTokenRequest here for the subsequent refresh. Otherwise you'd have to embed a new background thread within the the LocalMSALController if you wanted to do it all in there.

Reply:
Outdated